core/src/blocktree.rs

```diff
@@ -356,13 +371,40 @@ impl Blocktree {
             write_batch.put::<cf::ErasureMeta>((slot, set_index), &erasure_meta)?;
         }
 
+        batch_processor.write(write_batch)?;
+
         if should_signal {
-            for signal in self.new_blobs_signals.iter() {
+            for signal in &self.new_blobs_signals {
                 let _ = signal.try_send(true);
             }
         }
 
-        batch_processor.write(write_batch)?;
+        if self.completed_slots.len() > 0 && !newly_completed_slots.is_empty() {
+            let mut slots: Vec<_> = (0..self.completed_slots.len() - 1)
+                .map(|_| newly_completed_slots.clone())
+                .collect();
+
+            slots.push(newly_completed_slots);
```

Comment:
this is a nice optimization, but hard to read...  is newly_completed_slots large, typically?

Reply:
heh, highly doubt it, probably a few u64s at most. Pinching pennies xD